File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file199.f90
subroutine kernel_syrk(ni, nj, alpha, beta, c, a)
implicit none

DATA_TYPE, dimension(ni, ni) :: a
DATA_TYPE, dimension(nj, ni) :: c
DATA_TYPE :: alpha , beta
integer :: nj, ni
integer :: i, j, k

!$pragma scop
do i = 1, _PB_NI
do j = 1, _PB_NI
c(j, i) = c(j, i) * beta
end do
end do
do i = 1, _PB_NI
do j = 1, _PB_NI
do k = 1, _PB_NJ
c(j, i) = c(j, i) + (alpha * a(k, i) * a(k, j))
end do
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file199.cpp
#include <stdio.h>

#define DATA_TYPE int
#define _PB_NI ni
#define _PB_NJ nj

void kernel_syrk(int ni, int nj, DATA_TYPE alpha, DATA_TYPE beta, DATA_TYPE c[100][100], DATA_TYPE a[100][100]) 
{
    int i, j, k;
    
    for (i = 0; i < _PB_NI; i++)
        for (j = 0; j < _PB_NI; j++)
            c[j][i] *= beta;

    for (i = 0; i < _PB_NI; i++)
        for (j = 0; j < _PB_NI; j++)
            for (k = 0; k < _PB_NJ; k++)
                c[j][i] += alpha * a[k][i] * a[k][j];
}
